test-data/unit/check-functions.test

```diff
@@ -3694,3 +3694,17 @@ def defer() -> int: ...
 [out]
 main: note: In function "a":
 main:6: error: Unsupported operand types for + ("int" and "str")
+
+[case testNoExtraNoteForUnpacking]
+from typing import Protocol
+
+class P(Protocol):
+    arg: int
+    # Something that list and dict also have
+    def __contains__(self, item: object) -> bool: ...
+
+def foo(x: P, y: P) -> None: ...
+
+foo(*[0, ''])  # E: Argument 1 to "foo" has incompatible type "*list[object]"; expected "P"
+foo(**{'x': 0, 'y': ''})  # E: Argument 1 to "foo" has incompatible type "**dict[str, object]"; expected "P"
```

Comment:
```suggestion
foo(*cast(list[object], ...))  # E: Argument 1 to "foo" has incompatible type "*list[object]"; expected "P"
foo(**cast(dict[str, object], ...))  # E: Argument 1 to "foo" has incompatible type "**dict[str, object]"; expected "P"
```
maybe let's do^ in case we ever change join behaviour?

Reply:
Good idea, thanks! Added annotations instead of casts though